### PR TITLE
Add Rate Limit details

### PR DESCRIPTION
**Pull request by hardillb**
part of flowforge/flowforge#2530

## Description

Allow rate limiting to be enabled

## Related Issue(s)

flowforge/flowforge#2530

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: helm/flowforge/values.yaml [hunk 1]
```diff
@@ -36,6 +36,13 @@ forge:
   support:
     enabled: false
 
+  rate_limits:
+    enabled: false
+    global: false
+    max: 1000
+    maxAnonymous: 10
+    timeWindow: 60000
```
Comment (Steve-Mcl) [inline helm/flowforge/values.yaml:44]: Should these not match the public defaults we decided in the issue Ben?

If yes, then they should be...

 ```suggestion
  rate_limits:
    enabled: false - disabled by default.
    # global: true - A setting to enable limits across all routes
    # max: 1000 - the default rate limit is 1000 operations per route over the time window
    # timeWindow: '1 Minute' - or 60000
```

Comment (hardillb) [reply helm/flowforge/values.yaml]: These values need to be fully defined in this file, we can't comment them out, and `timeWindow` must be a number.

I've pushed an update to change global to true

File: helm/flowforge/README.md [hunk 2]
```diff
@@ -119,6 +119,16 @@ Enables FlowForge Telemetry
  - `forge.privateCA.configMapName` name of ConfigMap to store the CA Cert bundle (default `ff-ca-certs`)
  - `forge.privateCA.certs` base64 encoded CA certificate PEM bundle of trusted certificates. This needs to be generated without line breaks e.g. `base64 -w 0 certs.pem` (default not set)
  
+ ### Rate Limiting
+
+ - `forge.rate_limits.enabled` (default `false`)
+ - `forge.rate_limits.global` (default `false`)
+ - `forge.rate_limits.timeWindow` Time in milliseconds to evaluate requests over (default 60000)
+ - `forge.rate_limits.max` Max requests per timeWindow (default: 1000)
+ - `forge.rate_limits.maxAnonymous` Max anonymous requests per timeWindow (default: 10)
```
Comment (Steve-Mcl) [inline helm/flowforge/README.md]: Are these public defaults or our cloud solution defaults Ben?

If general release defaults, these should be...

```suggestion
 - `forge.rate_limits.enabled` (default `false`)
 - `# forge.rate_limits.global` (default `true`)
 - `# forge.rate_limits.timeWindow` Time in milliseconds to evaluate requests over (default 60000)
 - `# forge.rate_limits.max` Max requests per timeWindow (default: 1000)
 - `# forge.rate_limits.maxAnonymous` Max anonymous requests per timeWindow (optional)
```


From the docs...
<html>
<body>

flag | desc
-- | --
rate_limits.enabled | Enables rate limiting. Default: false
rate_limits.global | Enables rate limiting for all routes. Default: true (defaults to all routes being rate limited)
rate_limits.timeWindow | The time window in which requests are counted. Default: 60000 (1 minute)
rate_limits.max | The maximum number of requests allowed in the time window. Default: 1000
rate_limits.maxAnonymous | The maximum number of requests allowed in the time window for anonymous users. Default: not configured (defaults to rate_limits.max)


</body>
</html>

Comment (hardillb) [reply helm/flowforge/README.md]: We do not include comments in the docs!

Comment (hardillb) [reply helm/flowforge/README.md]: The value in the docs MUST match the values in the values.yml file. I will fix the global value

Comment (hardillb) [reply helm/flowforge/README.md]: Actually, no, the docs already match the values.

I took the defaults from the forge PR, so they are the actual defaults, we haven't set the production values (and these do NOT match what what you've asked for on staging)

Comment (Steve-Mcl) [reply helm/flowforge/README.md]: Im lost off Ben.  (probably my fault)

For general release, the conclusion was that `forge.rate_limits.global` would be `true` by default

Which docs show this as `false`?

The merged FF docs show it as true - https://github.com/flowforge/flowforge/blob/main/docs/install/configuration.md#rate-limiting-configuration

File: helm/flowforge/values.yaml [hunk 3]
```diff
@@ -36,6 +36,13 @@ forge:
   support:
     enabled: false
 
+  rate_limits:
+    enabled: false
+    global: true
+    max: 1000
+    maxAnonymous: 10
```
Comment (Steve-Mcl) [inline helm/flowforge/values.yaml:43]:
```suggestion
```

According to our defaults and docs, this should not be set.

Comment (Steve-Mcl) [reply helm/flowforge/values.yaml:43]: perhaps it could be `undefined` if it needs to be present?

Comment (hardillb) [reply helm/flowforge/values.yaml:43]: Having is as an example value in the flowforge.yml **STRONGLY** implies it is the default value. If that is not the case then it needs be updated.

Also it's not clear, is this a fastify-rate-limit value or something you've added?

**Review by Steve-Mcl — APPROVED** [review]

**Comment by Steve-Mcl** [comment]
@hardillb. approved. However I will not merge any of these - I will leave merging for you to coordinate appropriately.

File: helm/flowforge/templates/configmap.yaml [hunk 4]
```diff
@@ -163,3 +163,7 @@ data:
     branding:
 {{ toYaml .Values.forge.branding | indent 6 }}
     {{- end }}
+    {{- if .Values.forge.rate_limits }}
+    rate_limits:
+{{ toYaml .Values.forge.rate_limits | indent 6 }}
+    {{- end }}
```
Comment (hardillb) [inline helm/flowforge/templates/configmap.yaml:169]:
```suggestion
    {{- end }}

```

**Pull request merged**